File: parsers/go/binary_test.go
package synx

import "testing"

func TestBinaryStaticRoundtrip(t *testing.T) {
	r := Parse("name App\nport 8080\n")
	bytes, err := Compile(r, false)
	if err != nil {
		t.Fatalf("compile: %v", err)
	}
	if !IsSynxb(bytes) {
		t.Fatalf("magic missing")
	}
	restored, err := Decompile(bytes)
	if err != nil {
		t.Fatalf("decompile: %v", err)
	}
	if !Equal(r.Root, restored.Root) {
		t.Fatalf("roundtrip mismatch")
	}
}

func TestBinaryMagicCheck(t *testing.T) {
	if !IsSynxb([]byte{'S', 'Y', 'N', 'X', 'B', 1, 0}) {
		t.Fatalf("magic mismatch")
	}
	if IsSynxb([]byte{'J', 'S', 'O', 'N'}) {
		t.Fatalf("false positive")
	}
}

func TestBinaryInvalidMagicRejected(t *testing.T) {
	bad := make([]byte, 11)
	copy(bad, []byte("WRONG"))
	if _, err := Decompile(bad); err == nil {
		t.Fatalf("expected error")
	}
}

File: parsers/dart/test/binary_test.dart
import 'package:synx/synx.dart';
import 'package:test/test.dart';

void main() {
  test('static roundtrip', () {
    final r = parseFull('name App\nport 8080\n');
    final compiled = compile('name App\nport 8080\n');
    expect(compiled.ok, isTrue, reason: compiled.error);
    expect(isSynxb(compiled.bytes!), isTrue);
    final restored = decompile(compiled.bytes!);
    expect(restored.ok, isTrue, reason: restored.error);
    expect(restored.result!.root, equals(r.root));
  });

  test('magic check', () {
    expect(isSynxb([0x53, 0x59, 0x4E, 0x58, 0x42, 1, 0]), isTrue);
    expect(isSynxb([0x4A, 0x53, 0x4F, 0x4E]), isFalse);
  });

  test('invalid magic rejected', () {
    final bad = List<int>.filled(11, 0);
    bad[0] = 'W'.codeUnitAt(0);
    bad[1] = 'R'.codeUnitAt(0);
    bad[2] = 'O'.codeUnitAt(0);
    bad[3] = 'N'.codeUnitAt(0);
    bad[4] = 'G'.codeUnitAt(0);
    final r = decompile(bad);
    expect(r.ok, isFalse);
  });
}
